props/blaster.h

```diff
@@ -61,23 +45,74 @@ class Blaster : public PropBase {
   }
 
   bool auto_firing_ = false;
+  int shots_fired_ = 0;
+
+#ifdef BLASTER_SHOTS_UNTIL_EMPTY
+  const int max_shots_ = BLASTER_SHOTS_UNTIL_EMPTY;
+#else
+  const int max_shots_ = -1;
+#endif
 
   virtual void Fire() {
-    if (!SaberBase::IsOn()) return; // Don't allow firing when off.
+    if (max_shots_ != -1) {
+      if (shots_fired_ >= max_shots_) {
+        SaberBase::DoEmpty();
+        return;
+      }
+    }
 
     if (blaster_mode == MODE_AUTO) {
```

Comment:
Random thought: we could specify the chance of jamming in the config file.
Then we could use clash or shake gestures to trigger un-jamming. :)

Reply:
Already thought of that and was gonna add it but wanted to get this in first. But that is exactly how I wanted Jam/Unjam to work. Random chance to create a jam and shake to unjam.

Reply:
Btw, shake is a new gesture, so it's possible it just doesn't work very well.
You can use "monitor strokes" to see if it is working or not.

Reply:
I have triggered it with Clash and will circle back and try to add Shake but I think I have to as you say enable some of the gestures that may be turned off in the blaster due to the empty overrides (I had to add code back into the empty Clash override to make the clash work)